File: java/src/main/java/org/example/ConcreteStrategies.java
package org.example;

import lombok.Getter;

@Getter
class WhatsAppNotifier implements INotifier {
    private final Strategies id = Strategies.WHATSAPP;

    @Override
    public void send(Notification notification) {
        System.out.println("notificación enviada por " + this.id);
        System.out.println(notification);
    }
}

@Getter
class EmailNotifier implements INotifier {
    private final Strategies id = Strategies.EMAIL;

    @Override
    public void send(Notification notification) {
        System.out.println("notificación enviada por " + this.id);
        System.out.println(notification);
    }
}

@Getter
class PushNotifier implements INotifier {
    private final Strategies id = Strategies.PUSH;

    @Override
    public void send(Notification notification) {
        System.out.println("notificación enviada por " + this.id);
        System.out.println(notification);
    }
}

File: ts/ConcreteStrategies.ts
import { Strategies } from "./constants"
import { INotifer } from "./INotifer"
import { Notification } from "./Notification"

export class EmailNotifer implements INotifer {
    id: Strategies = Strategies.EMAIL
  send(notification: Notification): Promise<{ success: boolean; info: string; }> {
        console.log(notification)
        return Promise.resolve({ success: true, info: `Enviando por ${this.id}` })
    }
}

export class WhatsAppNotifer implements INotifer {
    id: Strategies = Strategies.WHATSAPP
    send(notification: Notification): Promise<{ success: boolean; info: string; }> {
        console.log(notification)
        return Promise.resolve({ success: true, info: `Enviando por ${this.id}` })
    }
}

export class PushNotifer implements INotifer {
    id: Strategies = Strategies.PUSH
    send(notification: Notification): Promise<{ success: boolean; info: string; }> {
        console.log(notification)
        return Promise.resolve({ success: true, info: `Enviando por ${this.id}` })
    }
}
